ui/src/logs/components/expandable_message/ExpandedContainer.tsx

```diff
@@ -0,0 +1,151 @@
+// Libraries
+import React, {Component, MouseEvent, CSSProperties} from 'react'
+
+// Components
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+
+// Types
+import {NotificationAction} from 'src/types'
+
+// Decorators
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface Props {
+  notify: NotificationAction
+  onClose: () => void
+  maxWidth: number
+  maxHeight: number
+  minWidth: number
+  minLeft: number
+  width: number
+  left: number
+  top: number
+  padding: number
+  children: JSX.Element
+}
+
+interface State {
+  scrollTop: number
+}
+
+const MIN_LEFT = 120
+
+@ErrorHandling
+export class ExpandedContainer extends Component<Props, State> {
+  private initialVisibility: boolean
+
+  constructor(props: Props) {
+    super(props)
+
+    this.initialVisibility = this.isVisible
+    this.state = {scrollTop: 0}
+  }
+
+  public componentDidMount() {
+    if (!this.initialVisibility) {
+      this.props.onClose()
```

Comment:
dose this mean there's a chance that it could appear on the initial render and then close?

Reply:
Good question, no initial appearance, this was a ux edge case I decided to tackle. The renderMessage prevents rendering an expanded message when the width is too small so there won't be any kind of flickered rendering.  This `onClose` prevents a sort of surprising user experience issue I'd like your thoughts on.

- User doesn't have enough room to display a message 
- User clicks on a message and nothing happens
- User resizes or scroll the message into view  
- User sees the expanded message they clicked  

----

> Without the onClose
> ![kapture 2018-11-01 at 10 42 57](https://user-images.githubusercontent.com/4994741/47858678-fe05ff80-ddc2-11e8-80b7-f0206e6cf2c7.gif)

> With the onClose
> ![kapture 2018-11-01 at 10 45 37](https://user-images.githubusercontent.com/4994741/47858887-6523b400-ddc3-11e8-8d54-5ea67b87c617.gif)

Reply:
I agree, i think the case where nothing happens seems weird.